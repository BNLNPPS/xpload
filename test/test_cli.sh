#!/usr/bin/env bash

SCRIPT_DIR=`dirname "$0"`
DATA_DIR=`xpl config path`
mkdir -p $DATA_DIR

echo $SCRIPT_DIR

dd if=/dev/random of=$DATA_DIR/payload_10.data bs=1k count=1 &> /dev/null
dd if=/dev/random of=$DATA_DIR/payload_11.data bs=1k count=1 &> /dev/null

ls -la $DATA_DIR/

md5sum $DATA_DIR/payload_10.data
md5sum $DATA_DIR/payload_11.data

xpl insert tag_1 domain_1 payload_10.data -s 3
xpl insert tag_1 domain_2 payload_11.data -s 5
xpl insert tag_2 domain_1 payload_10.data -s 7
xpl insert tag_2 domain_2 payload_11.data -s 9

rm -f /tmp/test_cli.out

{
xpl show tags
xpl show domains

xpl fetch tag_1 -d domain_1
xpl fetch tag_1 -d domain_2
xpl fetch tag_1 -s 3
xpl fetch tag_1 -s 5
xpl fetch tag_1 -s 3 -d domain_1
xpl fetch tag_1 -s 3 -d domain_2
} |& tee -a /tmp/test_cli.out


diff /tmp/test_cli.out $SCRIPT_DIR/test_cli.out
