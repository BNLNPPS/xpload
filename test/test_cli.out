Creating tag tag_1... ok
Creating tag tag_2... ok
Creating tag tag_3... ok
Copying payload file /tmp/path/to/payload/data1/domain_1/3176afe8baa90e9a6e488c67dde6e125_payload_00.data...
Copying payload file /tmp/path/to/payload/data1/domain_2/e88061f1bb6627db9e99c278f0cc7572_payload_01.data...
Copying payload file /tmp/path/to/payload/data1/domain_1/56ee22fc91d3576aa69547118bb65c5e_payload_10.data...
Copying payload file /tmp/path/to/payload/data1/domain_2/2aaf8466b7baa9333d9b083ad1dae031_payload_11.data...
Copying payload file /tmp/path/to/payload/data1/domain_3/139673081dcedf4f67f7248c94eca4e7_payload_N0.data...
tag_1
tag_2
tag_3
domain_1
domain_2
domain_3
/tmp/path/to/payload/data1/domain_1/3176afe8baa90e9a6e488c67dde6e125_payload_00.data
/tmp/path/to/payload/data1/domain_2/e88061f1bb6627db9e99c278f0cc7572_payload_01.data
/tmp/path/to/payload/data1/domain_1/3176afe8baa90e9a6e488c67dde6e125_payload_00.data
/tmp/path/to/payload/data1/domain_1/3176afe8baa90e9a6e488c67dde6e125_payload_00.data
/tmp/path/to/payload/data1/domain_2/e88061f1bb6627db9e99c278f0cc7572_payload_01.data
/tmp/path/to/payload/data1/domain_1/3176afe8baa90e9a6e488c67dde6e125_payload_00.data
Error: FileExistsError No payload file was found in any prefix
/tmp/path/to/payload/data1/domain_1/56ee22fc91d3576aa69547118bb65c5e_payload_10.data
/tmp/path/to/payload/data1/domain_2/2aaf8466b7baa9333d9b083ad1dae031_payload_11.data
Error: FileExistsError No payload file was found in any prefix
